### PR TITLE
Cut down loading time of handbook page #117

### LINKED ISSUE #117: Cut down loading time of handbook page
May be some form of incremental or on-demand loading instead of trying to load everything at once and freezing the page until everything is loaded?

### PR DESCRIPTION
**Pull request by acjh**
Fixes #117
[Preview](https://rawgit.com/acjh/website/117-cut-loading-time-handbook/contents/handbook.html)

See update below: [#issuecomment-266413313](https://github.com/nus-cs2103/website/pull/210#issuecomment-266413313)

### CODE REVIEW

**Comment by damithc** [comment]
Confirmed this is faster?

**Comment by acjh** [comment]
I've tested load-on-demand for every anchor in `#table-of-contents` on Chrome (Windows and Android).
Still trying to get these to work:
- [ ] Anchors to Appendix B and its subsections
- [ ] Anchors to Appendix C's subsections

**Comment by acjh** [comment]
All anchors in `#table-of-contents` now trigger load-on-demand by default.
- Load on demand: [`.../contents/handbook.html`](https://rawgit.com/acjh/website/117-cut-loading-time-handbook/contents/handbook.html)
- Load all incrementally: [`.../contents/handbook.html?preview=all`](https://rawgit.com/acjh/website/117-cut-loading-time-handbook/contents/handbook.html?preview=all)
- Preview a specific section: [`.../contents/handbook.html?preview=preliminaries`](https://rawgit.com/acjh/website/117-cut-loading-time-handbook/contents/handbook.html?preview=preliminaries)
> `preview` can be any of [these sections](https://github.com/acjh/website/blob/117-cut-loading-time-handbook/contents/handbook.html#L181), **without** the `handbook-` prefix:
>&nbsp;
`var sections = [`
`        'handbook-preliminaries',`
`        'handbook-lectures',`
`        'handbook-tutorials',`
`        'handbook-textBooks',`
`        'handbook-programmingLanguages',`
`        'handbook-project',`
`        'handbook-project-product',`
`        'handbook-project-scope',`
`        'handbook-project-constraints',`
`        'handbook-project-deliverables',`
`        'handbook-project-v00',`
`        'handbook-project-v01',`
`        'handbook-project-v02',`
`        'handbook-project-v03',`
`        'handbook-project-v04',`
`        'handbook-project-v05rc',`
`        'handbook-project-v05',`
`        'handbook-project-retrospective',`
`        'handbook-project-individualReport',`
`        'handbook-project-assessment',`
`        'handbook-project-supervision',`
`        'handbook-teams',`
`        'handbook-peerEvaluations',`
`        'handbook-tools',`
`        'handbook-exams',`
`        'handbook-gradeBreakdown',`
`        'handbook-participation',`
`        'handbook-appendixA-principles',`
`        'handbook-appendixB-policies',`
[~`        'handbook-policy',`~](https://github.com/nus-cs2103/website/pull/210/commits/2f1c2b8e179f018d221039d24867e20588d6bfc0)
`        'handbook-appendixC-faq',`
`        'handbook-appendixD-help',`
`        'handbook-appendixE-github',`
`        'handbook-appendixF-teamworkIssues'];`

**Comment by acjh** [comment]
To do:
- [ ] Use `location.hash` for section to preview, instead of `?preview=section-to-preview`.  
This is required for handbook links from schedule page to load, and avoids redundancy.
- [ ] Load all sections incrementally by default, instead of having to specify `?preview=all`?
Unless `location` contains a `hash`, which would mean preview. Seems more intuitive?

**Comment by acjh** [comment]
- Load sections incrementally: [`.../contents/handbook.html`](https://rawgit.com/acjh/website/117-cut-loading-time-handbook/contents/handbook.html) (default)
- Preview a specific section: [`.../contents/handbook.html#handbook-preliminaries`](https://rawgit.com/acjh/website/117-cut-loading-time-handbook/contents/handbook.html#handbook-preliminaries)
(scrolls into view; remaining anchors in table of contents are load-on-demand)
> The `hash` can be any of [these sections](https://github.com/acjh/website/blob/117-cut-loading-time-handbook/contents/handbook.html#L181):
>&nbsp;
`var sections = [`
`        'handbook-preliminaries',`
`        'handbook-lectures',`
`        'handbook-tutorials',`
`        'handbook-textBooks',`
`        'handbook-programmingLanguages',`
`        'handbook-project',`
`        'handbook-project-product',`
`        'handbook-project-scope',`
`        'handbook-project-constraints',`
`        'handbook-project-deliverables',`
`        'handbook-project-v00',`
`        'handbook-project-v01',`
`        'handbook-project-v02',`
`        'handbook-project-v03',`
`        'handbook-project-v04',`
`        'handbook-project-v05rc',`
`        'handbook-project-v05',`
`        'handbook-project-retrospective',`
`        'handbook-project-individualReport',`
`        'handbook-project-assessment',`
`        'handbook-project-supervision',`
`        'handbook-teams',`
`        'handbook-peerEvaluations',`
`        'handbook-tools',`
`        'handbook-exams',`
`        'handbook-gradeBreakdown',`
`        'handbook-participation',`
`        'handbook-appendixA-principles',`
`        'handbook-appendixB-policies',`
[~`        'handbook-policy',`~](https://github.com/nus-cs2103/website/pull/210/commits/2f1c2b8e179f018d221039d24867e20588d6bfc0)
`        'handbook-appendixC-faq',`
`        'handbook-appendixD-help',`
`        'handbook-appendixE-github',`
`        'handbook-appendixF-teamworkIssues'];`

**Comment by acjh** [comment]
@damithc Ready for review.

**Comment by acjh** [comment]
Loads sections asynchronously, cut loading time from 20-30s to 5-7s.
---
- Load sections asynchronously: [`.../contents/handbook.html`](https://rawgit.com/acjh/website/117-cut-loading-time-handbook/contents/handbook.html) (default)
- Preview a specific section: [`.../contents/handbook.html?preview=preliminaries`](https://rawgit.com/acjh/website/117-cut-loading-time-handbook/contents/handbook.html?preview=preliminaries)
(scrolls into view; remaining anchors in table of contents are load-on-demand)
> `preview` can be any of [these sections](https://github.com/acjh/website/blob/117-cut-loading-time-handbook/contents/handbook.html#L181), **without** the `handbook-` prefix:
>&nbsp;
`var sections = [`
`        'handbook-preliminaries',`
`        'handbook-lectures',`
`        'handbook-tutorials',`
`        'handbook-textBooks',`
`        'handbook-programmingLanguages',`
`        'handbook-project',`
`        'handbook-project-product',`
`        'handbook-project-scope',`
`        'handbook-project-constraints',`
`        'handbook-project-deliverables',`
`        'handbook-project-v00',`
`        'handbook-project-v01',`
`        'handbook-project-v02',`
`        'handbook-project-v03',`
`        'handbook-project-v04',`
`        'handbook-project-v05rc',`
`        'handbook-project-v05',`
`        'handbook-project-retrospective',`
`        'handbook-project-individualReport',`
`        'handbook-project-assessment',`
`        'handbook-project-supervision',`
`        'handbook-teams',`
`        'handbook-peerEvaluations',`
`        'handbook-tools',`
`        'handbook-exams',`
`        'handbook-gradeBreakdown',`
`        'handbook-participation',`
`        'handbook-appendixA-principles',`
`        'handbook-appendixB-policies',`
[~`        'handbook-policy',`~](https://github.com/nus-cs2103/website/pull/210/commits/2f1c2b8e179f018d221039d24867e20588d6bfc0)
`        'handbook-appendixC-faq',`
`        'handbook-appendixD-help',`
`        'handbook-appendixE-github',`
`        'handbook-appendixF-teamworkIssues'];`

**Comment by acjh** [comment]
@damithc Ready for review.

File: contents/handbook-appendixB-policies.html [hunk 1]
```diff
@@ -0,0 +1,172 @@
+<h3>Policy on following instructions</h3>
```
Comment (damithc) [inline contents/handbook-appendixB-policies.html]: Why is this page changed?

Comment (acjh) [inline contents/handbook-appendixB-policies.html]: `appendixB-policies` was an empty file, which meant loading the section did nothing.
Instead, users would have to select a subsection to load the common `handbook-policy`.
The change made this consistent with how the appendices work (especially Appendix C).

**Comment by acjh** [comment]
> [`preview`](https://github.com/nus-cs2103/website/pull/210#issuecomment-266413313) can also be any of [these subsections](https://github.com/acjh/website/blob/117-cut-loading-time-handbook/contents/handbook.html#L51), **without** the `handbook-` prefix:
>&nbsp;
`'policy-followingInstructions'`
`'policy-teamSize'`
`'policy-workDistribution'`
`'policy-validAbsences'`
`'policy-responseTime'`
`'policy-techHelp'`
`'policy-publishingSubmissions'`
`'policy-plagiarism'`
`'policy-reuse'`
`'policy-outsiderHelp'`
`'policy-submissionLength'`
`'handbook-faq-highWorkload'`
`'handbook-faq-beanCounting'`
`'handbook-faq-separateWebsite'`
`'handbook-faq-slideFormat'`
`'handbook-faq-selfStudy'`
`'handbook-faq-timelyArrival'`
`'handbook-faq-noLaptop'`
`'handbook-faq-narrowScope'`
`'handbook-faq-vagueRequirements'`
`'handbook-faq-favoriteTool'`
`'handbook-faq-manySubmissions'`
`'handbook-faq-intermediateMarks'`
`'handbook-faq-cs2101Differences'`

**Pull request merged**